lib/SocketHelpers.cpp

```diff
@@ -147,9 +162,23 @@ std::optional<Message> ReceiveProxyWifiMessage(const wil::unique_socket& socket)
 
 static void SendBytes(wil::unique_socket& socket, gsl::span<const uint8_t> dataToSend)
 {
-    // TODO: Is it needed to loop on a send?
     while (dataToSend.size_bytes() > 0)
     {
+        // Wait until the destination is ready to receive
+        const timeval timeout{0, 500000}; // 0.5 sec timeout
+        fd_set write_set{};
+        FD_SET(socket.get(), &write_set);
+        const auto socket_ready = select(0, nullptr, &write_set, nullptr, &timeout);
+        if (socket_ready == SOCKET_ERROR)
+        {
+            THROW_LAST_ERROR_MSG("Error while waiting to send a message");
+        }
+        else if (socket_ready == 0)
+        {
+            THROW_WIN32_MSG(ERROR_TIMEOUT, "Timeout while waiting to send a message");
+        }
```

Comment:
I see you've skipped checking `FD_ISSET(socket.get(), &write_set)` which seems fine since there's only one socket in the set, but should this verify that `socket_ready == 1` to be certain?

Reply:
What makes me slightly uneasy is that we're trusting the API to behave correctly in two instances: that when it returns `!= SOCKET_ERROR && !0`, that `FD_ISSET(...)` is true. A compromise might be to add an assert for one of these things. Anyway, this is a nit, so feel free to ignore.

Reply:
I think I'll ignore this. We have to trust the API to be respected, or we cannot do anything anyway (what if it says the socket is ready to receive, but actually it isn't??). That's never ending.

Reply:
That's fair enough.